frontend/hooks/data-fetching/course.ts

```diff
@@ -122,33 +111,28 @@ export async function sendMassInvites(
 }
 
 export async function getSemesters(): Promise<Semester[]> {
-    return doApiRequest("/semesters/")
+    return doApiRequest("/api/semesters/")
         .then((res) => res.json())
         .catch((_) => []);
 }
 
 export async function createTag(courseId: number, name: string): Promise<Tag> {
     const payload = { name };
 
-    return doApiRequest(`/courses/${courseId}/tags/`, {
+    return doApiRequest(`/api/courses/${courseId}/tags/`, {
         method: "POST",
         body: payload,
     })
         .then((res) => res.json())
         .catch((_) => null);
 }
 
-function newResourceFetcher<R>(path, ...args): R | Promise<R> {
-    return fetch(path, ...args).then((res) => res.json());
-}
-
 export const useQueues = (courseId: number, initialData: Queue[]) =>
-    useResourceListNew<Queue>(
+    useResourceList<Queue>(
         `/api/courses/${courseId}/queues/`,
         (id) => `/api/courses/${courseId}/queues/${id}/`,
         {
             initialData,
-            fetcher: newResourceFetcher,
             refreshInterval: QUEUE_STATUS_POLL_INTERVAL,
             refreshWhenHidden: true,
```

Comment:
You can also get rid of these refreshWhenHidden fields, since they are configured in the global SWRConfig. They were here before because of our peer dep issue resulting in SWRConfig not being able to configure rest-hooks from last time.